From be181e649ee90d387ce69f5c48531197188398d9 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Wed, 5 Jun 2019 10:12:03 +0200
Subject: GCC: avoid ambiguous NoDestructor creation in
 GetNeverSniffedMimeTypes.

Use brace-list notation to wrap the already existing brace-list for
initializing the flat-set. This resolves an ambiguous instantiation
in GCC.

Bug: 819294
Change-Id: I89ddf12522d62a5140a8c2c41dc98e30ec7a0e78

diff --git a/services/network/cross_origin_read_blocking.cc b/services/network/cross_origin_read_blocking.cc
index 91f5d21b6e5b..1da0c0e18101 100644
--- a/services/network/cross_origin_read_blocking.cc
+++ b/services/network/cross_origin_read_blocking.cc
@@ -213,7 +213,7 @@ bool ShouldEnforceInitiatorLock() {
 // confirmation sniffing because images, scripts, etc. are frequently
 // mislabelled by http servers as HTML/JSON/XML).
 base::flat_set<std::string>& GetNeverSniffedMimeTypes() {
-  static base::NoDestructor<base::flat_set<std::string>> s_types({
+  static base::NoDestructor<base::flat_set<std::string>> s_types{{
       // The list below has been populated based on most commonly used content
       // types according to HTTP Archive - see:
       // https://github.com/whatwg/fetch/issues/860#issuecomment-457330454
@@ -223,7 +223,7 @@ base::flat_set<std::string>& GetNeverSniffedMimeTypes() {
       "application/x-gzip",
       "application/x-protobuf",
       "application/zip",
-  });
+  }};
 
   // All items need to be lower-case, to support case-insensitive comparisons
   // later.
-- 
2.20.1

